CMakePresets.json

```diff
@@ -5,7 +5,9 @@
       "name": "_base",
       "hidden": true,
       "cacheVariables": {
-        "LV_CONF_SKIP": true
+        "LV_CONF_SKIP": true,
```

Comment:
Hey this is just for the default setup to be running without any issues. The user will probably remove it.
If i omit this option i cannot build lvgl directly outof the box because
`#include "lv_conf.h"` fails. Since we do not provide dummy config files for windows, linux etc. we must skip the config.
(or is there another way?)

Reply:
By default LVGL assumes `lv_conf.h` is placed in the same folder as `lvgl`. It provides a configuration free setting by default. Can we do it with `CMakePresets.json` too?

Reply:
sorry there might be an misunderstanding. The goal of this preset is to enable the user to build this project with MSVC without having to make any changes. I can use an existing lv_conf.h but I cannot find any lv_conf.h within the repo. (And I'm guessing there is none since lv_conf.h is application depended)

IMHO if we enforce the user to create an lv_conf.h first before building this would make first steps with lvgl much more complicated. 
=> IMHO goal should be that any MSVC user could clone this repo as it is and build it without having to make any changes. 
Or am I mistaken here?

Obviously when integrating you would like to use your own lv_conf.h.
Usually when working with CmakePresets you will create your own CmakePresets.json within your project which will overwrite the Presets from the lvgl repo. 
=> The presets from the lvgl repo are only intended as a quick start guide and will not be used in user projects.

To make things even more clear usally your project structure looks like this

- my_repo
-      CMakePresets.json
-      CMakeLists.txt
-      main.c
- lvgl
-     CMakepresets.json
-    ... lvgl sources

To build this via console you need to navigate to my_repo and invoke the presets
```
cd my_repo
cmake --preset my_project_configure_preset
cmake --build --preset my_project_build_preset
```
(or tell your IDE to use the CmakePresets from my_project)

as you can see the CmakePresets.json from the lvgl repo is not invoked at all :)

Did i make things more clear?

Reply:
Thank you for the detailed description! The goal is indeed to make getting started as simple as possible. Normally we create ready to use projects like [these](https://github.com/lvgl?q=lv_port_pc&type=all&language=&sort=)

> The goal of this preset is to enable the user to build this project with MSVC without having to make any changes. 

However, user still need to create a `main.c` which is way less trivial than creating `lv_conf.h`.

Reply:
hey, alright more clarification:)
You can build the lvgl repo now as a library without having to make any changes.

However in most cases you would like to build an executable (hence the main.c). 
In fact if you don't want to create an executable you do not need the main.c. You can also create a new wrapper library around lvgl. (In that case you obviously don't need a main.c)

So to sum up, this PR simply enables users to build this repo out of the box without having to make any changes. The provides cmake presets can be easily overwritten by any other cmake project.
For example: If i check out the https://github.com/lvgl/lv_port_pc_visual_studio
the CmakePresets.json within the lvgl repo does not have any effect because the .sln file from lv_port_pc_visual_studio does not invoke cmake :)

Reply:
So now the questions is which case is more common, and that one should be the default:
1. Build a library from LVGL
2. Build an executable with LVGL

I think 2) is more common therefore we shouldn't add `"LV_CONF_SKIP": true` by default but let the user add it if they are building a library. Note that even when compiling a library you can choose to use `lv_cong.h`.

I'm afraid that having `"LV_CONF_SKIP": true,` by default will confuse most of the users as they won't know why their changes in `lv_conf.h`have no effect. Having a compiler error about "`lv_conf.h` is missing" is more explicit.

Reply:
Hey, I agree the second option is more common.
However in that case the project will not build since the lv_conf.h is not found.
I'd suggest to add a demo executable to this project (i.e. by using cmakes add_executable()). This demo exe can be build with a demo lv_conf.h
Since all is located in a "demo" folder within this project it is clear to every new user that this needs to be adapted when creating an own project. Our cmake presets will simply build this demo exe as well.
Or do you have a better suggestion how to handle the lv_conf.h. We need to provide one at build time, otherwise build will fail :)

However the `LV_SKIP_CONFIG: true` is already in the master branch. To keeps things clear I'd suggest to create a separate PR for this topic. Let's keep things in this PR simple and straight forward :)

Reply:
> However the LV_SKIP_CONFIG: true is already in the master branch. To keeps things clear I'd suggest to create a separate PR for this topic. Let's keep things in this PR simple and straight forward :)

Ah, you are right. In this case it's really unrelated to this PR. Let's keep it as it is :smile: